ApiFirst/AppServices/Services/PortfolioAppService.cs

```diff
@@ -0,0 +1,190 @@
+﻿using AppModels.AppModels.Orders;
+using AppModels.AppModels.Portfolios;
+using AppModels.EnumModels;
+using AppServices.Interfaces;
+using AutoMapper;
+using DomainModels.Models;
+using DomainServices.Interfaces;
+using EntityFrameworkCore.UnitOfWork.Factories;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Transactions;
+
+namespace AppServices.Services
+{
+    public class PortfolioAppService : IPortfolioAppService
+    {
+        private readonly IPortfolioService _portfolioService;
+        private readonly IMapper _mapper;
+        private readonly ICustomerBankInfoAppService _customerBankInfoAppService;
+        private readonly IProductAppService _productAppService;
+        private readonly IOrderAppService _orderAppService;
+        private readonly IPortfolioProductService _portfolioProductService;
+
+        public PortfolioAppService(
+            IPortfolioService portfolio,
+            IMapper mapper,
+            ICustomerBankInfoAppService customerBankInfoAppServices,
+            IProductAppService productAppServices,
+            IOrderAppService orderAppServices,
+            IPortfolioProductService portfolioProductServices)
+        {
+            _portfolioService = portfolio ??
+                throw new ArgumentNullException(nameof(portfolio));
+            _mapper = mapper ??
+                throw new ArgumentNullException(nameof(mapper));
+            _customerBankInfoAppService = customerBankInfoAppServices ??
+                throw new ArgumentNullException(nameof(customerBankInfoAppServices));
+            _productAppService = productAppServices ??
+                throw new ArgumentNullException(nameof(productAppServices));
+            _orderAppService = orderAppServices ??
+                throw new ArgumentNullException(nameof(orderAppServices));
+            _portfolioProductService = portfolioProductServices ??
+                throw new ArgumentNullException(nameof(portfolioProductServices));
+        }
+
+        public async Task<long> CreateAsync(CreatePortfolio model)
+        {
+            var portfolio = _mapper.Map<Portfolio>(model);
+            return await _portfolioService.CreateAsync(portfolio).ConfigureAwait(false);
+        }
+
+        public async Task<IEnumerable<PortfolioResponse>> GetAllAsync()
+        {
+            var result = await _portfolioService.GetAllAsync();
```

Comment:
E nos outros métodos...
```suggestion
            var result = await _portfolioService.GetAllAsync().ConfigureAwait(false);
```